docs/emitters/code-generation/server/server-code-generation.md

```diff
@@ -0,0 +1,375 @@
+
+                /// <inheritdoc />
+                public override Task<IActionResult> CreateComment([Required] string project, [Required] string todoItem, [Required] CreateCommentRequest body)
+                {
+                    var newComment = new Comment()
+                    {
+                        Id = Guid.NewGuid().ToString(),
+                        ProjectId = project,
+                        TodoItemId = todoItem,
+                        Body = body.Body,
+                        Created = DateTime.Now
+                    };
+                    _comments.Add(newComment);
+                    return Task.FromResult<IActionResult>(CreatedAtAction(nameof(GetComment), new { id = newComment.Id }, newComment));
+                }
+
+                /// <inheritdoc />
+                public override Task<IActionResult> GetComment(string id)
+                {
+                    var comment = _comments.FirstOrDefault(c => c.Id == id);
+                    if (comment == null)
+                    {
+                        return Task.FromResult<IActionResult>(NotFound());
+                    }
+
+                    return Task.FromResult<IActionResult>(Ok(comment));
+                }
+            }
+        }
+        ```
+
+        **Important**: if your controller file is not automatically generated as `CommentOpsOperationsController.cs`, you will have to manually add in `[ApiController]` and inherit the base class `CommentOpsOperationsControllerBase`.
+
+*   **Node.js/express example:**
+
+    [Placeholder: Detailed instructions on setting up a Node.js/Express project, integrating the generated code, and running the server will be added here.]
+
+    1.  **Create a basic express app:** Use the Express generator to create a new project:
+
+        ```bash
+        npm install -g express-generator
+        express my-express-app
+        cd my-express-app
+        npm install
+        ```
+
+    2.  **Install additional dependencies:** You may need additional dependencies depending on your API requirements (e.g., `body-parser` for parsing request bodies).
+
+        ```bash
+        npm install body-parser --save
+        ```
+
+    3.  **Integrate generated code:** Copy the contents of the `servers/node/tsp-output/@typespec/http-server-javascript` directory into your Express project. You will likely need to adapt the generated code to fit the Express routing and middleware structure.
+
+    4.  **Configure routes:** Define your API routes in your Express app, mapping them to the generated controller functions.
+
+    5.  **Run the server:** Start the Express server:
+
+        ```bash
+        npm start
+        ```
+
+## Cli: running the project and adding necessary code:
+
+Now that you have the base controller, models and service interface generated, the next step is to implement the code with a service scaffolding and business logic for a runnable server. Please follow the instruction below:
```

Comment:
This has changed quite a bit

Reply:
@markcowl where/how can I get the latest info on how it works?

Reply:
revised